setup.py

```diff
@@ -1,27 +1,76 @@
-from setuptools import setup
+# -*- coding: utf-8 -*-
+from __future__ import print_function
+from setuptools import setup, find_packages
+from setuptools.command.test import test as TestCommand
+import codecs
+import os
+import sys
+import re
 
 
+def read(*parts):
+    here = os.path.abspath(os.path.dirname(__file__))
+    # intentionally *not* adding an encoding option to open
+    return codecs.open(os.path.join(here, *parts), 'r').read()
+
+
+def find_version(*file_paths):
+    version_file = read(*file_paths)
+    version_match = re.search(r"^__version__ = ['\"]([^'\"]*)['\"]",
+                              version_file, re.M)
+    if version_match:
+        return version_match.group(1)
+    raise RuntimeError("Unable to find version string.")
+
+
+class PyTest(TestCommand):
+    user_options = [('pytest-args=', 'a', "Arguments to pass to py.test")]
+
+    def initialize_options(self):
+        TestCommand.initialize_options(self)
+        self.pytest_args = []
+
+    def finalize_options(self):
+        TestCommand.finalize_options(self)
+        self.test_args = []
+        self.test_suite = True
+
+    def run_tests(self):
+        #import here, cause outside the eggs aren't loaded
+        import pytest
+        errno = pytest.main(self.pytest_args)
+        sys.exit(errno)
+
 setup(
     name='envparse',
-    version='0.1.6',
+    version=find_version('.', 'envparse.py'),
     url='https://github.com/rconradharris/envparse',
     license='MIT',
     author='Rick Harris',
     author_email='rconradharris@gmail.com',
-    description='Simple Environment Variable Parsing',
-    long_description=__doc__,
+    tests_require=['pytest'],
+    install_requires=[''],
```

Comment:
Not sure, I've seen it written like that in official docs but then it could probably be omitted as well.